runelite-client/src/main/java/net/runelite/client/plugins/agility/Obstacles.java

```diff
@@ -90,145 +93,7 @@
-		// Heroes Guild
-		CREVICE_9739, CREVICE_9740,
-		// Fossil Island
-		HOLE_31481, HOLE_31482, LADDER_30938, LADDER_30939, LADDER_30940, LADDER_30941, ROPE_ANCHOR, ROPE_ANCHOR_30917,
-		RUBBER_CAP_MUSHROOM,
-		ROCKS_31757, ROCKS_31758, ROCKS_31759, PILLAR_31809,
-		// West Brimhaven
-		ROPESWING_23568, ROPESWING_23569,
-		// Brimhaven Dungeon
-		VINE_26880, VINE_26882, PIPE_21728, STEPPING_STONE_19040, PIPE_21727, LOG_BALANCE_20882, LOG_BALANCE_20884,
-		STEPPING_STONE_21738, STEPPING_STONE_21739, TIGHTGAP,
-		// Lumbridge
-		STILE_12982,
-		// Edgeville Dungeon
-		MONKEYBARS_23566, OBSTACLE_PIPE_16511,
-		// Miscellania
-		STEPPING_STONE_11768,
-		// Kalphite
-		CREVICE_16465,
-		// Eagles' Peak
-		ROCKS_19849,
-		// Catherby
-		CROSSBOW_TREE_17062, ROCKS_17042,
-		// McGrubor's Woods
-		LOOSE_RAILING,
-		// Cairn Isle
-		ROCKS_2231,
-		// South Kourend
-		STEPPING_STONE_29728, STEPPING_STONE_29729, STEPPING_STONE_29730,
-		// Cosmic Temple
-		JUTTING_WALL_17002,
-		// Arandar
-		ROCKS_16514, ROCKS_16515, LOG_BALANCE_3933,
-		// South River Salve
-		STEPPING_STONE_13504,
-		DARK_TUNNEL_10047,
-		// Ectofuntus
-		WEATHERED_WALL, WEATHERED_WALL_16526,
-		// Mos Le'Harmless
-		STEPPING_STONE_19042,
-		// North River Salve
-		ROCKS_16998, ROCKS_16999, ORNATE_RAILING, ORNATE_RAILING_17000,
-		// West Zul-Andra
-		STEPPING_STONE_10663,
-		// Yanille Agility Dungeon
-		BALANCING_LEDGE_23548, OBSTACLE_PIPE_23140, MONKEYBARS_23567, PILE_OF_RUBBLE_23563, PILE_OF_RUBBLE_23564,
-		// High Level Wilderness Dungeon
-		CREVICE_19043,
-		// Revenant Caves
-		PILLAR_31561,
-		// Elf Camp Isafdar Tirranwn
-		LOG_BALANCE_3931, LOG_BALANCE_3930, LOG_BALANCE_3929, LOG_BALANCE_3932, DENSE_FOREST_3938, DENSE_FOREST_3939,
-		DENSE_FOREST_3998, DENSE_FOREST_3999, DENSE_FOREST, LEAVES, LEAVES_3924, LEAVES_3925, STICKS, TRIPWIRE,
-		// Gu'Tanoth bridge
-		GAP, GAP_2831,
-		// Lumbridge Swamp Caves
-		STEPPING_STONE_5948, STEPPING_STONE_5949, ROCKS_6673,
-		// Morytania Pirate Ship
-		ROCK_16115,
-		// Agility Pyramid Entrance
-		CLIMBING_ROCKS_11948, CLIMBING_ROCKS_11949,
-		// Lumber Yard
-		BROKEN_FENCE_2618,
-		// Ungael and Vorkath crater
-		NULL_25337, NULL_29868, NULL_29869, NULL_29870, ICE_CHUNKS_31822, NULL_31823, ICE_CHUNKS_31990,
-		// Underwater Area Fossil Island
-		TUNNEL_30959, HOLE_30966, OBSTACLE, OBSTACLE_30767, OBSTACLE_30964, OBSTACLE_30962,
-		// Tree Gnome Village
-		LOOSE_RAILING_2186,
-		// Weiss
-		LITTLE_BOULDER, ROCKSLIDE_33184, ROCKSLIDE_33185, NULL_33327, NULL_33328, LEDGE_33190, ROCKSLIDE_33191, FALLEN_TREE_33192,
-		// Al-Kharid
-		BROKEN_WALL_33344, BIG_WINDOW,
-		// Burgh de Rott
-		LOW_FENCE,
-		// Taverley
-		STILE,
-		// Asgarnian Ice Dungeon
-		STEPS,
-		// Fossil Island Wyvern Cave
-		STAIRS_31485,
-		// Mount Karuulm
-		ROCKS_34397, ROCKS_34396
-	);
+	static final Set<Integer> SHORTCUT_OBSTACLE_IDS = Stream.of(AgilityShortcut.values()).map(value -> value.getObstacleIds()).flatMap(l -> l.stream()).collect(Collectors.toSet());
```

Comment:
I would probably get rid of this and just reference the map to be added to the obstacle class.